src/web/rustdoc.rs

```diff
@@ -259,110 +293,113 @@ impl RustdocPage {
         self,
         rustdoc_html: &[u8],
         max_parse_memory: usize,
-        req: &mut Request,
+        templates: &TemplateData,
+        metrics: &Metrics,
+        config: &Config,
         file_path: &str,
-    ) -> IronResult<Response> {
-        use iron::{headers::ContentType, status::Status};
-
-        let templates = req
-            .extensions
-            .get::<super::TemplateData>()
-            .expect("missing TemplateData from the request extensions");
-
-        let metrics = req
-            .extensions
-            .get::<crate::Metrics>()
-            .expect("missing Metrics from the request extensions");
-
+    ) -> AxumResult<AxumResponse> {
         let is_latest_url = self.is_latest_url;
         let canonical_url = self.canonical_url.clone();
 
         // Build the page of documentation
-        let ctx = ctry!(req, tera::Context::from_serialize(self));
-        let config = extension!(req, Config);
+        let ctx = tera::Context::from_serialize(self).context("error creating tera context")?;
+
         // Extract the head and body of the rustdoc file so that we can insert it into our own html
         // while logging OOM errors from html rewriting
         let html = match utils::rewrite_lol(rustdoc_html, max_parse_memory, ctx, templates) {
             Err(RewritingError::MemoryLimitExceeded(..)) => {
                 metrics.html_rewrite_ooms.inc();
 
-                let err = anyhow!(
-                    "Failed to serve the rustdoc file '{}' because rewriting it surpassed the memory limit of {} bytes",
-                    file_path, config.max_parse_memory,
-                );
-
-                ctry!(req, Err(err))
+                return Err(AxumNope::InternalError(
```

Comment:
since `AxumNope::InternalError` also does `report_error`, this slightly changes behaviour since previously these errors didn't end up in sentry. 

Looking at the metrics, this didn't happen for a long time (not once in the last 30 days), 

so I would keep the behaviour as-is and then decide if we want to mute this error in sentry.

Reply:
This is left over from before we used lol-html and it was happening nearly every day. We can probably delete it today as long is it doesn't fire in the next week, lol-html uses a streaming design that rarely uses even 5 MB at once.